consumer/src/main/kotlin/streams/StreamsSinkConfigurationListner.kt

```diff
@@ -0,0 +1,146 @@
+package streams
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import org.apache.commons.configuration2.ImmutableConfiguration
+import org.neo4j.kernel.internal.GraphDatabaseAPI
+import org.neo4j.logging.Log
+import org.neo4j.plugin.configuration.ConfigurationLifecycleUtils
+import org.neo4j.plugin.configuration.EventType
+import org.neo4j.plugin.configuration.listners.ConfigurationLifecycleListener
+import streams.procedures.StreamsSinkProcedures
+import streams.service.TopicUtils
+import streams.utils.Neo4jUtils
+import streams.utils.StreamsUtils
+
+class StreamsSinkConfigurationListener(private val db: GraphDatabaseAPI,
+                                       private val log: Log) : ConfigurationLifecycleListener {
+
+    private val mutex = Mutex()
+
+    var eventSink: StreamsEventSink? = null
+
+    private val streamsTopicService = StreamsTopicService()
+
+    private fun shutdown() {
+        val isShuttingDown = eventSink != null
+        if (isShuttingDown) {
+            log.info("[Sink] Shutting down the Streams Sink Module")
+        }
+        eventSink?.stop()
+        eventSink = null
+        StreamsSinkProcedures.unregisterStreamsEventSink(db)
```

Comment:
I understand this only shuts down the SINK.    What about the SOURCE?

What if I configure those dynamically?

Additionally -- if we do make the source dynamic in the config as well (which we should) then we need to consider this scenario:

1. User is producing -> Kafka
2. User changes some sink parameter (fine)
3. During the restart of the plugin, a TX comes through.
4. Did we miss producing the TX because we were stopped while it happened?  Or do we block the TX until we're ready to produce it out?

Reply:
It shouts down both, please look at this class:
https://github.com/neo4j-contrib/neo4j-streams/pull/394/files#diff-4b0c66acb7d47e7d66f7c655ffffc1672fbc070f9058880bf0c7bfdc5a8715d5R14

Yes at this very moment we restart both but we can add a mechanism in order to check differences in order to check if source/sink properties are changed and then restart the module only if there is a property change for that module.

Reply:
When the source is restarted we unplug the transaction handler so the database keeps working but we lost what it's doing during the restart